src/domain/occurrences/occurrencesTable/ActionsDropdown.tsx

```diff
@@ -0,0 +1,239 @@
+  const {
+    focusedIndex,
+    setup: setupKeyboardNav,
+    teardown: teardownKeyboardNav,
+  } = useKeyboardNavigation({
+    container: container,
+    listLength: 3,
+  });
+
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  const toggleMenu = React.useCallback(() => {
+    setIsMenuOpen(!isMenuOpen);
+  }, [isMenuOpen]);
+
+  const ensureMenuIsOpen = React.useCallback(() => {
+    if (!isMenuOpen) {
+      setIsMenuOpen(true);
+    }
+  }, [isMenuOpen]);
+
+  const isComponentFocused = React.useCallback(() => {
+    const active = document.activeElement;
+
+    return active instanceof Node && container.current?.contains(active);
+  }, []);
+
+  const onDocumentClick = (event: MouseEvent) => {
+    const target = event.target;
+
+    // Close menu when clicking outside of the component
+    if (!(target instanceof Node && container.current?.contains(target))) {
+      setIsMenuOpen(false);
+    }
+  };
+
+  const onDocumentFocusin = React.useCallback((event: FocusEvent) => {
+    const target = event.target;
+
+    if (!(target instanceof Node && container.current?.contains(target))) {
+      setIsMenuOpen(false);
+    }
+  }, []);
+
+  const setFocusToButton = () => {
+    toggleButton.current?.focus();
+  };
+
+  const onKeyDown = React.useCallback(
+    (event: KeyboardEvent) => {
+      if (!isComponentFocused()) return;
+
+      switch (event.key) {
+        case 'ArrowUp':
+          ensureMenuIsOpen();
+          break;
+        case 'ArrowDown':
+          ensureMenuIsOpen();
+          break;
+        case 'Enter':
+          if (isMenuOpen) {
+            toggleMenu();
+            setFocusToButton();
+            event.preventDefault();
+          }
+          break;
+        case 'Escape':
+          if (isMenuOpen) {
+            setIsMenuOpen(false);
+            setFocusToButton();
+          }
+          break;
+        case 'Tab':
+          setIsMenuOpen(false);
+      }
+    },
+    [ensureMenuIsOpen, isComponentFocused, isMenuOpen, toggleMenu]
+  );
+
+  React.useEffect(() => {
+    setupKeyboardNav();
+    document.addEventListener('click', onDocumentClick);
+    document.addEventListener('focusin', onDocumentFocusin);
+    document.addEventListener('keydown', onKeyDown);
+
+    // Clean up event listener to prevent memory leaks
+    return () => {
+      teardownKeyboardNav();
+      document.removeEventListener('click', onDocumentClick);
+      document.removeEventListener('focusin', onDocumentFocusin);
+      document.removeEventListener('keydown', onKeyDown);
+    };
+  }, [onDocumentFocusin, onKeyDown, setupKeyboardNav, teardownKeyboardNav]);
+
+  const goToEnrollmentsPage = () => {
+    history.push(
+      `/${locale}${ROUTES.ENROLLMENTS.replace(':id', eventId).replace(
+        ':occurrenceId',
+        row.id
+      )}`
+    );
+  };
+
+  const goToEditOccurrencePage = () => {
+    history.push(
+      `/${locale}${ROUTES.EDIT_OCCURRENCE.replace(':id', eventId).replace(
```

Comment:
Would this again work without the ${locale}?

Reply:
The url blinks if using url without locale, because it first goes to /events/palvelutarjotin:afzdfpmkpa/occurrences/T2NjdXJyZW5jZU5vZGU6NA==/edit and after that react-router redirects to /fi/events/palvelutarjotin:afzdfpmkpa/occurrences/T2NjdXJyZW5jZU5vZGU6NA==/edit
It was little bit irritating to me.

Reply:
ah ok 👍